006-strimzi-ui.md

```diff
@@ -0,0 +1,178 @@
+When I navigate to the topic listing
+And I filter for topic 'SampleTopic'
+Then I am told topic 'SampleTopic' does not exist in the cluster
+```
+
+#### Follow on capabilities
+
+Additional Kafka capabilities which could be added later include (but would not be limited to):
+
+- Create, update and delete topics on the topics list page (recontextualised as a Topics page - as per mock up)
+- View Consumer group status for a given topic, or the whole cluster
+- View the Brokers in a given cluster, their configuration, and where appropriate, allow modification of broker configuration
+- Provide details of Kafka listener/bootstrap addresses, along with sample configuration, to allow streamline client connectivity
+
+If desired, additional Strimzi provided capabilities, such as Kafka Users, could be leveraged/added/managed through this UI also (for example, allowing access to the UI and its capabilities, general CRUD updates to them etc). 
+
+Finally, one capability which may be of interest (which may have bearing on how this is surfaced in Strimzi itself) is having a single UI, which can manage multiple Kafka clusters deployed via the Strimzi operator.
+
+In all of these cases, capabilities can be added in a prioritised order, and should be added in a phased manner themselves (for example, add a view of the brokers and their configuration, then the ability to modify select configuration).
+
+#### Proposed deployment
+
+This UI could be deployed as a part of Strimzi as follows:
+
+![Suggested deployment](./images/006-deployment.png)
+
+Where:
+
+- The UI is deployed standalone, akin to how the Kafka bridge, Kafka Connect are currently for example (so a new CRD would be required, with configuration needed to reference which cluster(s) to interact with).
+- The UI's spec would contain 0 to N Kafka clusters the UI will operate against, which will: 
+  - Contain 1 or more 'backend' entries. These represent backend sources of data which can be surfaced in the UI.
+    - One of these entries will be the `strimzi-api` (ie each Kafka kind will contain its own api server).
+  - Each of these entries will define items such as the address to use to connect to it, any auth credentials required, Transport security required, etc.
+- The UI's spec will contain general configuration for the UI - such as certificates to expose to clients etc.
+
+This could look as follows in a CR configured by a user (note that common fields such as image, or readiness and liveness probes have been omitted here for conciseness, but would be present in a full CR). Comments are included inline for clarification of intent/type:
+
+```
+...
+spec:
+  clusters: # Array/list
+  - name: 'dev-cluster' # string - metadata, but could be used in the UI (eg for cluster selection if more than one cluster) in the clusters array (order of which respected by the UI)
+    clientCert: <mounted TLS cert to serve to clients> # optional - if omitted, UI server will serve via http rather than https
```

Comment:
so AFAIU this is the certificate for TLS encryption between a client (browser) and the UI server.
Why is this a field of a specific cluster? The UI server is just one (for showing more clusters data) so I guess it should be a root field directly under `spec`. It doesn't bring "information" related to connection between the UI server and a specific cluster.

Reply:
Tying into this, how would we want to limit some users from seeing details from one cluster and not another? Is this the purpose of the field or should this this just be at the root as Paolo suggests?

Reply:
Correct @ppatierno - it should be a level up. Will correct this.

@samuel-hawker indeed - we may need to suppress some details from some users. Alot of this depends on how authentication and resource access will work - ie a user would be associated with cluster A and B, but not C say. The `name` could be used here for this.